Content.Server/PowerSink/PowerSinkSystem.cs

```diff
@@ -2,12 +2,32 @@
 using Content.Server.Power.Components;
 using Content.Shared.Examine;
 using Robust.Shared.Utility;
+using Content.Server.Chat.Systems;
+using Content.Server.Station.Systems;
+using Robust.Shared.Timing;
 
 namespace Content.Server.PowerSink
 {
     public sealed class PowerSinkSystem : EntitySystem
     {
+        /// <summary>
+        /// Percentage of battery full to trigger the announcement warning at.
+        /// </summary>
+        private const float WarningMessageThreshold = 0.85f;
+
+        private readonly float[] WarningSoundThresholds = new float[] { .90f, .95f, .98f };
+
+        /// <summary>
+        /// Length of time to delay explosion from battery full state -- this is used to play
+        /// a brief SFX winding up the explosion.
+        /// </summary>
+        /// <returns></returns>
+        private readonly TimeSpan ExplosionDelayTime = new TimeSpan(0, 0, 0, 1, 465);
+
         [Dependency] private readonly ExplosionSystem _explosionSystem = default!;
+        [Dependency] private readonly IEntitySystemManager _entitySystemManager = default!;
+        [Dependency] private readonly SharedAudioSystem _audio = default!;
+        [Dependency] private readonly IGameTiming _gameTiming = default!;
 
         public override void Initialize()
         {
```

Comment:
You should subscribe to entity paused changes and offset explosiontime by the paused time so if the map gets paused it's still correct.

e.g. something like

SubscribeLocalEvent<PowerSinkComponent, EntityUnpausedEvent>(OnUnpaused);

private void OnUnpaused(EntityUid uid, PowerSinkComponent component, ref EntityUnpausedEvent)
{
	if (component.ExplosionTime == null)
		return;

	component.ExplosionTime += args.PausedTime;
}

Reply:
According to the comment on CurTime on IGameTiming, it is already affected by Paused.  Is this not correct?

Reply:
RT supports pausing individual maps so CurTime will always keep running and the relevant code needs to handle unpausing if it has timers.